src/metabase/query_processor/streaming/common.clj

```diff
@@ -112,3 +113,64 @@
       (if (and is-currency? (::mb.viz/currency-in-header merged-settings true))
         (str column-title " (" (currency-identifier merged-settings) ")")
         column-title))))
+
+(defn normalize-keys
+  "Update map keys to remove namespaces from keywords and convert from snake to kebab case."
+  [m]
+  (update-keys m (fn [k] (-> k name (str/replace #"_" "-") keyword))))
+
+(def col-type
+  "The dispatch function logic for format format-timestring.
+  Find the highest type of the object."
+  (some-fn :semantic_type :effective_type :base_type))
+
+(defmulti global-type-settings
+  "Look up the global viz settings based on the type of the column. A multimethod is used because they match well
+  against type hierarchies."
+  (fn [col _viz-settings] (col-type col)))
+
+(defmethod global-type-settings :type/Temporal [_ {::mb.viz/keys [global-column-settings] :as _viz-settings}]
+  (:type/Temporal global-column-settings {}))
+
+(defmethod global-type-settings :type/Date [_ {::mb.viz/keys [global-column-settings] :as _viz-settings}]
+  (merge
+    (:type/Temporal global-column-settings {})
+    {::mb.viz/time-enabled nil}))
+
+(defmethod global-type-settings :type/Time [_ {::mb.viz/keys [global-column-settings] :as _viz-settings}]
+  (merge
+    (:type/Temporal global-column-settings {::mb.viz/time-style "h:mm A"})
+    {::mb.viz/date-style ""}))
+
+(defmethod global-type-settings :type/DateTime [_ {::mb.viz/keys [global-column-settings] :as _viz-settings}]
+  (:type/Temporal global-column-settings {}))
+
+(defmethod global-type-settings :type/Number [_ {::mb.viz/keys [global-column-settings] :as _viz-settings}]
+  (:type/Number global-column-settings {}))
+
+(defmethod global-type-settings :type/Currency [_ {::mb.viz/keys [global-column-settings] :as _viz-settings}]
+  (:type/Currency global-column-settings {::mb.viz/number-style "currency"}))
+
+(defmethod global-type-settings :default [_ _viz-settings]
+  {})
+
+(defn viz-settings-for-col
+  "Get the unified viz settings for a column based on the column's metadata (if any) and user settings (⚙)."
+  [{column-name :name metadata-column-settings :settings :keys [field_ref] :as col} viz-settings]
+  (let [[_ field-id-or-name] field_ref
+        all-cols-settings (-> viz-settings
+                              ::mb.viz/column-settings
+                              ;; update the keys so that they will have only the :field-id or :column-name
+                              ;; and not have any metadata. Since we don't know the metadata, we can never
+                              ;; match a key with metadata, even if we do have the correct name or id
```

Comment:
This part is debatable right? Like, in a Slack thread I thought it was discussed that column name or field-id is not technically sufficient to disambiguate columns in all cases. 

This assumption (dropping metadata) has been the behaviour already I'm pretty sure, so this PR shouldn't try to fix it, but this is something we should revisit later maybe? Please disregard this if I'm misremembering.